### ISSUE TITLE
Trouble parsing has_one relationships

### ISSUE THREAD

**Issue by sirvine** [JSONAPI-Resources/jsonapi-resources#134]
I'm getting error code 115 on all POST requests that have has_one relationships.  I've traced the issue to the following code from [here](https://github.com/cerebris/jsonapi-resources/blob/4b837118cb1e10612d21f709bb0d2be9c451a531/lib/jsonapi/request.rb#L239):

```
   if !raw.is_a?(Hash) || raw.length != 2 || !(raw.has_key?('type') && raw.has_key?('id'))
      raise JSONAPI::Exceptions::InvalidLinksObject.new
    end
```

Looking at my Rails logs, the inbound params for a has_one component is in the following form `{"linkage"=>{"id"=>"24", "type"=>"suppliers"}}`, which is failing the last three conditionals above.

Update:

I assume this is related to https://github.com/json-api/json-api/pull/447 and https://github.com/cerebris/jsonapi-resources/issues/115

**Comment by fotinakis** [comment]
Just hit this too.

**Comment by vrybas** [comment]
@sirvine Fixed by #140 ?

**Comment by sirvine** [comment]
@vrybas I think so. It's been a year since I opened this, and I haven't encountered this issue since. Closing now.
